src/PluginClient.php

```diff
@@ -62,9 +65,20 @@ public function __construct($client, array $plugins = [], array $options = [])
      */
     public function sendRequest(RequestInterface $request)
     {
-        $promise = $this->sendAsyncRequest($request);
+        if (!($this->client instanceof HttpClient)) {
```

Comment:
The emulation add a non necessary layer of code in this case. 

As we will have a emulated sync call, which transform promise into a response / exception which is then transformed into promise, and then retransformed into response / exception.

Here we only do the conversion from promise to response / exception once.